ocf_data_sampler/torch_datasets/datasets/pvnet_uk.py

```diff
@@ -168,6 +166,36 @@ def get_gsp_locations(gsp_ids: list[int] | None = None) -> list[Location]:
     return locations
 
 
+def validate_channels_from_config(config: Configuration) -> None:
+    """Validate that channels in config have corresponding normalisation constants.
+    
+    Args:
+        config: Configuration object containing NWP and satellite channel information
+        
+    Raises:
+        ValueError: If there's a mismatch between configured channels and normalisation constants
+    """
+    # Validate NWP channels if NWP is configured
+    if hasattr(config.input_data, "nwp"):
+        for nwp_key, nwp_config in config.input_data.nwp.items():
+            provider = nwp_config.provider
+            validate_channels(
+                data_channels=nwp_config.channels,
+                means_channels=NWP_MEANS[provider].channel.values,
+                stds_channels=NWP_STDS[provider].channel.values,
+                source_name=provider
+            )
+
+    # Validate satellite channels if satellite is configured
+    if hasattr(config.input_data, "satellite"):
+        validate_channels(
+            data_channels=config.input_data.satellite.channels,
+            means_channels=RSS_MEAN.channel.values,
+            stds_channels=RSS_STD.channel.values,
+            source_name="satellite"
+        )
+
```

Comment:
Refactor to two functions
- `validate_nwp_channels_from_config`
- `validate_satellite_channels_from_config`
then you can use `validate_nwp_channels_from_config`